pkg/catalog/routes.go

```diff
@@ -514,9 +529,150 @@ func (mc *MeshCatalog) listMeshServices() []service.MeshService {
 	return services
 }
 
+func (mc *MeshCatalog) destinationServicesFromTrafficTarget(t *target.TrafficTarget) ([]service.MeshService, error) {
+	sa := service.K8sServiceAccount{
+		Name:      t.Spec.Destination.Name,
+		Namespace: t.Spec.Destination.Namespace,
+	}
+	destServices, err := mc.GetServicesForServiceAccount(sa)
+	if err != nil {
+		return nil, errors.Errorf("Err finding Services for Service Account %#v: %v", sa, err)
+	}
+	return destServices, nil
+}
+
+func (mc *MeshCatalog) getHostnamesBasedOnNamespace(meshService service.MeshService) ([]string, []string, error) {
+	svc := mc.kubeController.GetService(meshService)
+	if svc == nil {
+		return nil, nil, errors.Errorf("Error fetching service %q", meshService)
+	}
+
+	hostnamesSameNamespace := kubernetes.GetLocalNamespaceHostnamesForService(svc)
+	hostnamesDiffNamespace := kubernetes.GetNamespaceScopedHostnamesForService(svc)
+
+	return hostnamesSameNamespace, hostnamesDiffNamespace, nil
```

Comment:
just out of curiosity: why do we want to differentiate local and namespaced hostnames ?

Reply:
Great question. We'll use the InboundTrafficPolicy and OutboundTrafficPolicy objects to program virtual hosts. Because a domain has to be unique across virtual hosts, we'll want to create different policy objects for local hostnames vs namespaced hostnames so that we don't have to do any additional computation in rds/response.go at the time we're actually programming the virtual hosts.

Reply:
+1, why can't this function automatically add the additional hostnames `service, service:port` as is done by `GetHostnamesForUpstreamService`?